src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AWS/SQS/CreateQueueAsyncIntegration.cs

```diff
@@ -0,0 +1,72 @@
+    [InstrumentMethod(
+        AssemblyName = "AWSSDK.SQS",
+        TypeName = "Amazon.SQS.AmazonSQSClient",
+        MethodName = "CreateQueueAsync",
+        ReturnTypeName = "System.Threading.Tasks.Task`1<Amazon.SQS.Model.CreateQueueResponse>",
+        ParameterTypeNames = new[] { "Amazon.SQS.Model.CreateQueueRequest", ClrNames.CancellationToken },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "3.*.*",
+        IntegrationName = AwsConstants.IntegrationName)]
+    public class CreateQueueAsyncIntegration
+    {
+        private const string Operation = "CreateQueue";
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TCreateQueueRequest">Type of the request object</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method</param>
+        /// <param name="request">The request for the SQS operation</param>
+        /// <param name="cancellationToken">CancellationToken value</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TCreateQueueRequest>(TTarget instance, TCreateQueueRequest request, CancellationToken cancellationToken)
+            where TCreateQueueRequest : ICreateQueueRequest, IDuckType
+        {
+            if (request.Instance is null)
+            {
+                return CallTargetState.GetDefault();
+            }
+
+            var scope = AwsSqsCommon.CreateScope(Tracer.Instance, $"{AwsConstants.AwsService}.{Operation}", out AwsSqsTags tags);
+            tags.Operation = Operation;
+            tags.Service = AwsConstants.AwsService;
+            tags.QueueName = request.QueueName;
+
+            return new CallTargetState(scope);
+        }
+
+        /// <summary>
+        /// OnAsyncMethodEnd callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TResponse">Type of the response, in an async scenario will be T of Task of T</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="response">Response instance</param>
+        /// <param name="exception">Exception instance in case the original code threw an exception.</param>
+        /// <param name="state">Calltarget state value</param>
+        /// <returns>A response value, in an async scenario will be T of Task of T</returns>
+        public static TResponse OnAsyncMethodEnd<TTarget, TResponse>(TTarget instance, TResponse response, Exception exception, CallTargetState state)
+            where TResponse : ICreateQueueResponse
+        {
+            if (state.Scope?.Span.Tags is AwsSqsTags tags)
+            {
+                tags.QueueUrl = response.QueueUrl;
+            }
+
+            state.Scope.DisposeWithException(exception);
```

Comment:
I think we're missing a null propagator? Though not _all_ our integrations do this, so I'm not 100%. But if not, then the null propagator on the previous `if()` doesn't make sense 😉 

```suggestion
            state.Scope?.DisposeWithException(exception);
```

Reply:
I followed up on another comment, but the `DisposeWithException` extension method does a null check on the `Scope` argument. Would it be more readable to do the null check outside the extension method and then have the extension method expect an non-null value?

Reply:
Ah, good point. Well, IMO the null check at the callsite is better for readability and expectations, but having it in the extension method is safer!  My biggest issue is with inconsistency, so I think we leave it as-is, and do a small cleanup to make all the invocations identical (i.e. drop the null check from callsites)

Reply:
Cleanup done in separate PR: https://github.com/DataDog/dd-trace-dotnet/pull/1493